packages/plugin-commands-installation/src/recursive.ts

```diff
@@ -438,7 +438,7 @@ function calculateRepositoryRoot (
   // assume repo root is workspace dir
   let relativeRepoRoot = '.'
   for (const rootDir of projectDirs) {
-    const relativePartRegExp = new RegExp(`^(\\.\\.\\${path.sep})+`)
+    const relativePartRegExp = new RegExp(`^(\\.\\.\\${path.sep})*(\\.\\.)?`)
```

Comment:
I am not sure why this was allowed at all. It doesn't make sense to store the workspace manifest in a subdirectory. This way pnpm will not know that it should run in the context of a workspace.

Reply:
that's what rush does @zkochan, it stores the workspace manifest in common/temp/pnpm-workspace.yaml. The original PR for this change was: https://github.com/pnpm/pnpm/pull/2532